src/Service.Host/client/src/components/StockPools.js

```diff
@@ -0,0 +1,309 @@
+                { fieldName: 'defaultLocation', searchResultFieldName: 'locationId' },
+                { fieldName: 'defaultLocationName', searchResultFieldName: 'locationCode' }
+            ],
+            clearSearch: clearStorageLocations,
+            renderCell: searchRenderCell
+        },
+        { field: 'sequence', editable: true, headerName: 'Sequence', width: 100 }
+    ];
+
+    const sortedStockPools = stockPools?.slice().sort((a, b) => {
+        const fa = a.sequence;
+        const fb = b.sequence;
+
+        if (fa === null && fb !== null) {
+            return 1;
+        }
+        if (fa !== null && fb === null) {
+            return -1;
+        }
+        if (fa === null && fb === null) {
+            return 0;
+        }
+        return fa - fb;
+    });
+
+    return (
+        <Page homeUrl={config.appRoot} showAuthUi={false}>
+            {stockPoolColumns?.filter(c => c.type === 'search').map(c => renderSearchDialog(c))}
+            <Grid container spacing={3}>
+                <Grid size={12}>
+                    <Typography variant="h4">Stock Pools</Typography>
+                </Grid>
+                {(isLoading ||
+                    accountingCompanyLoading ||
+                    updateLoading ||
+                    createStockPoolLoading) && (
+                    <Grid size={12}>
+                        <List>
+                            <Loading />
+                        </List>
+                    </Grid>
+                )}
+                <Grid size={12}>
+                    <DataGrid
+                        getRowId={row => row.id}
+                        rows={sortedStockPools}
+                        editMode="cell"
+                        processRowUpdate={processRowUpdate}
+                        columns={stockPoolColumns}
+                        rowHeight={34}
+                        loading={false}
+                        hideFooter
+                        isCellEditable={params => {
+                            if (params.field === 'stockPoolCode') {
+                                return params.row?.creating === true;
+                            }
+                            return true;
+                        }}
+                    />
+                </Grid>
+                <Grid item xs={4}>
+                    <Button onClick={addNewRow} variant="outlined" disabled={creating}>
+                        Add Stock Pool
+                    </Button>
+                </Grid>
+                <Grid item xs={4}>
+                    <Button
+                        onClick={() => {
+                            stockPools
+                                .filter(sp => sp.updated === true)
+                                .forEach(sp => {
+                                    const updatedStockPool = stockPools.find(
+                                        pool => pool.stockPoolCode === sp.stockPoolCode
+                                    );
```

Comment:
redundant? sp is already the current stockPool in the context of this forEach loop, so no need to employ another find-loop to obtain it